### PR TITLE
DO-13 initial ci workflow

### PR DESCRIPTION
**Pull request by dkravetz**
### References

[DO-13](https://xainag.atlassian.net/browse/DO-13)

### Summary

Create initial CI Workflow/Pipeline

### Are there any open tasks/blockers for the ticket?

Figure out the correct order to avoid tool conflicts (isort, black, etc)

---

### Reviewer checklist

*Reviewer agreement:*

* Reviewers assign themselves at the start of the review.
* Reviewers do **not** commit or merge the merge request.
* Reviewers have to check and mark items in the checklist.

**Merge request checklist**

- [x] Conforms to the merge request title naming `XP-XXX <a description in imperative form>`.
- [x] Each commit conforms to the naming convention `XP-XXX <a description in imperative form>`.
- [x] Linked the ticket in the merge request title or the references section.
- [x] Added an informative merge request summary.

**Code checklist**

- [x] Conforms to the branch naming `XP-XXX-<a_small_stub>`.
- [x] Passed scope checks.
- [x] Added or updated tests if needed.
- [x] Added or updated code documentation if needed.
- [x] Conforms to Google docstring style.
- [x] Conforms to XAIN structlog style.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    branches:
+      - master
+jobs:
+  main:
+    runs-on: ubuntu-18.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Setup Python 3.7
```
Comment (atymoshchuk) [inline .github/workflows/main.yml]: Haven't we decided that our main Python version is 3.6?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+          isort --check-only --indent=4 -rc setup.py conftest.py benchmarks \
+          examples xain_fl
+
+      - name: pylint
+        run: |
+          pylint --rcfile=pylint.ini benchmarks examples xain_fl
+
+      - name: mypy
+        run: |
+          mypy benchmarks examples/* xain_fl
```
Comment (atymoshchuk) [inline .github/workflows/main.yml]: can we allow mypy to fail for now?

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+      - name: Sphinx check
+        env:
+          SPHINXOPTS: "-W"
+        run: |
+          cd docs/
+          make html
```
Comment (atymoshchuk) [inline .github/workflows/main.yml]: Please use `make docs` command

File: .github/workflows/main.yml (hunk 1 above)
Comment (dkravetz) [reply .github/workflows/main.yml]: Aren't we going for the lowest possible version on the SDK?

Comment (atymoshchuk) [reply .github/workflows/main.yml]: Yes, which is 3.6, in setup.py we have `python_requires=">=3.6"`

File: .github/workflows/main.yml (hunk 2 above)
Comment (dkravetz) [reply .github/workflows/main.yml]: done

File: .github/workflows/main.yml (hunk 3 above)
Comment (dkravetz) [reply .github/workflows/main.yml]: done

File: .github/workflows/main.yml (hunk 1 above)
Comment (dkravetz) [reply .github/workflows/main.yml]: Done. Please create a ticket to update to Python 3.7

**Pull request merged**